From dae65a5636c58dfd3e1ec3cc7a55eb127dd6a4f3 Mon Sep 17 00:00:00 2001
From: Pan Xiuli <xiuli.pan@linux.intel.com>
Date: Wed, 24 Feb 2021 14:46:38 +0000
Subject: [PATCH 2/2] ASoC: SOF: sof-pci-dev: use sof_suspend as shutdown
 callback

According to hardware spec and PMC FW requirment, the DSP must be
in D3 state before entering S5. use sof_suspend as shutdown callback
here to make sure DSP is in D3 state

Signed-off-by: Pan Xiuli <xiuli.pan@linux.intel.com>
---
 sound/soc/sof/sof-pci-dev.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sound/soc/sof/sof-pci-dev.c b/sound/soc/sof/sof-pci-dev.c
index 1f1fccd68cca..0a739876758c 100644
--- a/sound/soc/sof/sof-pci-dev.c
+++ b/sound/soc/sof/sof-pci-dev.c
@@ -376,6 +376,11 @@ static void sof_pci_remove(struct pci_dev *pci)
 	pci_release_regions(pci);
 }
 
+static void sof_pci_shutdown(struct pci_dev *pci)
+{
+	snd_sof_suspend(&pci->dev);
+}
+
 /* PCI IDs */
 static const struct pci_device_id sof_pci_ids[] = {
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_MERRIFIELD)
@@ -444,6 +449,7 @@ static struct pci_driver snd_sof_pci_driver = {
 	.id_table = sof_pci_ids,
 	.probe = sof_pci_probe,
 	.remove = sof_pci_remove,
+	.shutdown = sof_pci_shutdown,
 	.driver = {
 		.pm = &sof_pci_pm,
 	},
-- 
2.25.1

